validator/validator.bal

```diff
@@ -62,3 +131,168 @@ public isolated function validate(json|anydata data, typedesc<anydata>? targetFH
 isolated function validateFhirResource(anydata data) returns r4:FHIRValidationError? {
     return validate(data);
 }
+
+isolated function parseConstraintErrors(string message) returns string[] {
+
+    string[] errors = [];
+
+    string:RegExp regex = re `\n`;
+    string[] data = regex.split(message);
+
+    //Parsing for dateTime errors.
+    foreach var i in 0 ... data.length() - 1 {
+        regexp:Groups[] invalidDates = re `\$\.([\w\.\[\]]+):pattern`.findAllGroups(data[i]);
```

Comment:
Why is this validation only related to dates?

Reply:
Because currently constraints are only used for date-time validation, other validations are done by the parser.

> Why is this validation only related to dates?